### PR TITLE
Modal: full feature support with Dimmer and docs

### LINKED ISSUE #175: Modal Keyboard Event Handlers
Would be helpful to add these key actions to the Modal component as default behaviors using the `onKeyUp` event handler, as opposed to having to add them every time a Modal component is used.
- [ ] `[ESC]` hides modal

### PR DESCRIPTION
**Pull request by levithomason**
An `image content` Modal requires the image to be a `div.ui.image > img`.  However, some components, like the List, require markup of `img.ui.image`.

Image updates in #280 provide a `wrapped` prop for choosing the div wrapped markup variation.
## Updates

Fixes #175 #84.  This PR adds many features and full docs for the Modal.  Added:
- missing ModalDescription
- portal render to body
- optional configurable dimmer
- auto scrolling when height is larger than window
- close on esc (fixes #175)
- close on click outside
- callbacks onHide, onShow

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -61,7 +61,6 @@ config = Object.assign({}, config, {
   compiler_output_path: paths.base(config.dir_docs_dist),
   compiler_public_path: __BASE__ || '/',
   compiler_vendor: [
-    'bluebird',
```
Comment (levithomason) [inline config.js:66]: 🎉

**Comment by tslater** [comment]
Will this let me have a dimmer/dimmed background? I can't figure out how to do that currently.

**Comment by levithomason** [comment]
Yes sir it will!  It includes every variation of the dimmer as well, even `false` for none.  There are also complete docs for this.

**Comment by tslater** [comment]
That will be awesome! Nice work!

**Comment by levithomason** [comment]
Just pushed 100% test coverage, will merge and release if it passes.

**Comment by levithomason** [comment]
@tslater I just released this in `stardust@0.33.0`.  The doc site should be updated by CI in just a few minutes.  Once it is, you can see the examples here:

http://technologyadvice.github.io/stardust/modules/modal

**Comment by tslater** [comment]
I just tested it out and it works great!

I did have to run `npm i --save semantic-ui-css`...because it said it was missing a package  `semantic-ui-css/something-or-other-about-forms`. 

The update also solved an issue where I had to add a top style to my modal, I deleted the rule and things are working great. Thanks so much!

**Comment by levithomason** [comment]
Thanks for the report.  There is one remaining jQuery plugin, form validation.  We are working to remove it, see #247.  I removed the `semantic-ui-css` dep which was a mistake as it also included the jQuery plugin.  Just released a patch in `v0.34.1`.

Glad to hear it is solving problems for you as well!

**Comment by tslater** [comment]
I just upgraded to 0.34.1 and removed semantic-ui-css. It's working great!

**Pull request merged**